src/game-logic/reducers/harvestPlot.js

```diff
@@ -96,3 +85,40 @@ export const harvestPlot = (state, x, y) => {
     },
   }
 }
+
+function harvestWeeds(state, x, y) {
+  const row = state.field[y]
+  const crop = row[x]
+  const item = itemsMap[crop.itemId]
+  const harvestedQuantity = getHarvestedQuantity(state)
+
+  state = removeFieldPlotAt(state, x, y)
+  state = addItemToInventory(state, item, harvestedQuantity)
+
+  return state
+}
+
+/**
+ * @param {farmhand.state} state
+ * @param {number} x
+ * @param {number} y
+ * @returns {farmhand.state}
+ */
+export const harvestPlot = (state, x, y) => {
+  const row = state.field[y]
+  const crop = row[x]
+
+  if (!crop) return state
+
+  if (
+    getPlotContentType(crop) === itemType.CROP &&
+    getCropLifeStage(crop) === GROWN &&
+    doesInventorySpaceRemain(state)
+  ) {
+    return harvestCrops(state, x, y)
+  } else if (getPlotContentType(crop) === itemType.WEEDS) {
```

Comment:
We'll want to check that `doesInventorySpaceRemain(state)` is `true` here as well so that weeds aren't pulled when there is no inventory space available to store them in.

Reply:
i pulled it out to be a guard clause since it applies to both scenarios